### ISSUE TITLE
Special situation of US minor IC changing to major IC

### ISSUE THREAD

**Issue by panther2** [triplea-maps/world_war_ii_global#4]
In the special case that the US decide to enter a state of war at the beginning of the US Collect Income Phase on US turn 3, the minor factories in  Eastern United States, Central United States and Western United States should convert to major factories at the same time (and not at the beginning of the US Combat Move Phase in the next US turn).

(The issue that the USA may choose to stay out of war at that point is already in the game notes.)

**Comment by simon33-2** [comment]
But the only issue caused is the incorrect warning about too many units bought USA4.

If we wanted to fix that properly, we'd change it so that the warning was to ask if you wanted to declare war and reject the purchase if you didn't.

**Comment by panther2** [comment]
IMHO the incorrect warning is a consequence of that issue. 
The timing of major IC switching to minor IC as happebing now in TripleA is not according to the rules - so this is the issue.
Your idea is a workaround, of course, but not a fix to become rules-compliant.

**Comment by simon33-2** [comment]
Actually, what I was saying would make it rules compliant. Notice I said to reject the purchase if they don't want to DOW which resolves the issues remaining with purchasing too many.

**Comment by simon33-2** [comment]
There are also incorrect warnings bought on other rounds where the DOW is intended and more than 9 units are bought. Also, allowing you to buy too many units or not place bought units are other problems. Most of these problems could be solved by changing the order to Combat Move before Purchase/Repair but that would not be (strictly) rules compliant. Change the rules, I say.

**Comment by panther2** [comment]
Never change the rules, I say ;-)
Or change the rules, but do it on different maps (or game.xml), then.

Again, my concern with this issue is not that warning. My concern is the correct handling of the process.

In the given case it would be:

1) On turn 3 at the beginning of the US Collect Income Phase let the US decide to enter a state of war or not.

2a) If they decide going to war immediately change the minor IC to major IC, proceed with collecting income.

2b) If they decide not to go to war proceed with collecting income.

And regarding those incorrect warnings let us open another issue.

**Comment by simon33-2** [comment]
So you aren't worried about the warnings for buying too many units on a USA1-3 combat move DOW, only on the special DOW at the end of the income phase? Why?

I'm thinking about adding a map with the combat movement before purchase/repair. That way, if you don't want to use it, you don't have to.

**Comment by panther2** [comment]
> So you aren't worried about the warnings for buying too many units on a USA1-3 combat move DOW, only on the special DOW at the end of the income phase? Why?

It was you who added the aspects of the warnings. When I opened this issue I was only talking about the timing of minors changing to majors in that special situation - as explained here: 
https://github.com/triplea-maps/world_war_ii_global/issues/4#issuecomment-244534585

Actually I am not concerned about the warnings because they are absolutely obsolete:

According to the rules in the Puchase Unit Phase you are allowed to buy as many units **as you can afford**. So technically there is no need to check, whether those units can be placed, so you don't need a warning. According to the rules this check takes place in the Unit Placement Phase.  If you bought more units than you can place, you get your money back for those units, that cannot be placed.

Maybe those warnings derive from an edition where overpurchased units have been treated differently. But that would be another map, then.

So concerning the warnings I would be fine with an Issue called "remove overbuy warnings in Purchase Unit Phase". 

> I'm thinking about adding a map with the combat movement before purchase/repair. That way, if you don't want to use it, you don't have to.

Sounds good to me.

**Comment by simon33-2** [comment]
Yeah, but Kreighund has clarified that the rule about refunding units not able to be placed (which aren't actually supposed to be held over) is only supposed to be used for inadvertent over purchasing. It's not supposed to be able to be gamed by buying a lot of inf and some more expensive units then placing units based on the results of battles.

**Comment by panther2** [comment]
Yes correct, I am aware of this. But the warning itself would not prevent from an intentional overbuy, but only confirm the buying player's intention. The opponent would not see it in the moment it is committed. 

So the warning for the buying player is pointless, too. 

It would be up to the opponent to object, when noticing an intentional overbuy. In a forum game that would become notable by the time of the turn post. The players would probably start a discussion about an intentional overbuy, then.

**Comment by simon33-2** [comment]
You could add a map property to the engine "MustPlace" or similar. So then you would force placements and have to prompt for DOW if more than 9 units bought.

I just don't see this as an important issue. It can be player enforced.

**Comment by panther2** [comment]
Yes.

**Comment by simon33-2** [comment]
There is one problem with changing the order - if you are going to land on a newly built CV triple A won't allow you to attack with air.

**Comment by panther2** [comment]
Yes, the engine can't know, that the purchase of the Carrier will occur after the combat move of the Planes.
Napoleonic Empire is a map, where combat move comes before purchase, IIRC. But there are no comparable units, I suppose.

**Comment by panther2** [comment]
Update:
While the discussion about "Overpurchsae" has become obsolete due to various discussions in the TripleA main repository, the original issue is still valid.
On A&A .org an addition to the game notes has been suggested, until this might get fixed.

**Comment by simon33-2** [comment]
It is possible to do a partial fix just by changing the point that the units upgrade to be before Purchase.

The trouble is that there is still an issue if Japan DOWs on UK+Anzac. Then USA can DOW on the Axis but only after Purchase and the warning will be wrongly triggered. I can't see a solution for that scenario, barring moving the politics to before purchase, which I'm not sure I would recommend.
